pallets/gear/src/lib.rs

```diff
@@ -635,7 +641,7 @@ pub mod pallet {
         /// Emits the following events:
         /// - `InitMessageEnqueued(MessageInfo)` when init message is placed in the queue.
         #[pallet::weight(
-            T::WeightInfo::submit_program(code.len() as u32, init_payload.len() as u32)
+            <T as pallet::Config>::WeightInfo::submit_program(code.len() as u32, init_payload.len() as u32)
```

Comment:
In my opinion it'd be better to use the unqualified `<T as Config>::WeightInfo` here (as long as the purpose is to resolve conflicts with other pallets' `WeightInfo` associated type). Not insisting though

Reply:
Thank you for clarification, fixed it.